src/modules/form/components/DynamicForm.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Grid, Button, Stack } from '@mui/material';
+import React, { useState } from 'react';
+
+import { FormDefinition, Item } from '../types';
+
+import DynamicField from './DynamicField';
+
+interface Props {
+  definition: FormDefinition;
+  onSubmit: (values: Record<string, any>) => void;
+}
+
+const DynamicForm: React.FC<Props> = ({ definition, onSubmit }) => {
```

Comment:
This is the component for dynamically rendering a form based on a form definition, which looks like this: https://github.com/greenriver/hmis-frontend/blob/46726c11749f49b3d8bb8767f027b500a89e00af/src/modules/form/data/intake.json

The linked definition looks like this rendered with DynamicForm:
![hmis dev test_5173_intake](https://user-images.githubusercontent.com/5333982/180514609-9fac88ab-bf0e-48a4-8d86-83c542d08fb7.png)